### PR TITLE
GH-96975: Skip incomplete frames in `PyEval_GetFrame`

### PR DESCRIPTION
**Pull request by brandtbucher**
* Issue: gh-96975

### CODE REVIEW

File: Programs/_testembed.c
Comment (ericsnowcurrently) [inline Programs/_testembed.c]: Drop the "return"?

```suggestion
    allocator->free(allocator->ctx, ptr);
```

File: Programs/_testembed.c
Comment (ericsnowcurrently) [inline Programs/_testembed.c]: I noticed you aren't unwrapping the allocator before finalizing.  That's probably okay though, given how `_testembed` works.  I was just thinking of how folks sometimes look to our test code for examples of how to do things.  It probably wouldn't be that hard.  \<nudge/>\<nudge/>

File: Programs/_testembed.c
Comment (ericsnowcurrently) [inline Programs/_testembed.c]:
```suggestion
test_get_incomplete_frame(void)
```

File: Python/ceval.c
Comment (ericsnowcurrently) [inline Python/ceval.c:6533]: Would it work to use `PyThreadState_GetFrame()`?  There are a couple slight differences,  but mostly it's the same.

```suggestion
    PyThreadState *tstate = _PyThreadState_GET();
    PyFrameObject *f = PyThreadState_GetFrame(tstate);
```

or just

```c
PyFrameObject *
PyEval_GetFrame(void)
{
    PyThreadState *tstate = _PyThreadState_GET();
    return PyThreadState_GetFrame(tstate);
}
```

**Review by ericsnowcurrently — APPROVED** [review]
Mostly LGTM

I've left a few minor comments but leave it to you to decide.

(I'm approving under the assumption you will get it right before merging.)

File: Python/ceval.c
Comment (brandtbucher) [reply Python/ceval.c:6533]: `PyThreadState_GetFrame` returns a new reference, but `PyEval_GetFrame` returns a borrowed one. We could `Py_XDECREF` the result before returning, but that sort of feels icky to me (the only reason it's safe is because we know how `PyThreadState_GetFrame` works).

**Review by ericsnowcurrently — APPROVED** [review]
LGTM

**Pull request merged**
